platforms/android/app/src/debug/java/com/tns/NativeScriptSyncServiceSocketImpl.java

```diff
@@ -175,6 +179,11 @@ public void run() {
                         if(runtime != null && doRefreshInt == DO_REFRESH_VALUE) {
                             runtime.runScript(new File(NativeScriptSyncServiceSocketImpl.this.context.getFilesDir(), "internal/livesync.js"), false);
```

Comment:
All such calls to runtime should be replaced with messages to the ForegroundService and then to the Activity and maybe await a response. Especially if we find we still need the runtime.lock/runtime.unlock before writing the a file.